README.md

```diff
@@ -170,155 +170,50 @@
-
-- #### V1.FLOW_ENDED
-  indicate that flow was finished
-
-  - Request payload
-
-    type: FlowEndedRequest  
-    object id of the finished flow instance  
-    ```typescript
-    {
-      flowInstanceId: string;
-    }
-    ```
-
-  - Response payload
-
-    type: void  
-
-### FLOW-RUNTIME Version1 events
-
-- #### V1.FLOWS.REQUIRE_CONTEXT  
-  Must be sent on application startup to get initial application context from the shell
-
-  - Request payload
-
-    type: object
-    ```typescript
-    {
-      clientIdentifier: string;
-      clientSecret: string;
-    }
-    ```
-
-  - Response payload
-
-    type: object
-    ```typescript
-    {
-      authToken: string;
-      cloudHost: string;
-      account: string;
-      accountId: string;
-      company: string;
-      companyId: string;
-      selectedLocale: string;
-      user: string;
-      userId: string;
-      initialContext: [
-        {
-          name: string;
-          type: string;
-          value: any;
-        }
-      ];
-      screenConfiguration: Object | null;
-    }
-    ```
-
-- #### V1.FLOWS.CAN_CONTINUE  
-  Used to indicate if current flow step ready to continue to the next step. Shell host will
-  block switching to the next step until this event sent with true value in payload. If data
-  at some point becames invalid again, this event with false value in payload can be sent
-  in order to block switching to next step again.
-
-  - Request payload
-
-    type: boolean
-
-  - Response not needed
-
-- #### V1.FLOWS.ON_CONTINUE
-  Will be sent from the shell host to flow app when user continue to the next flow step. Flow app should save(if needed) it's data, prepare output variables and send it in response to this event.
-
-  - Request payload
-
-    type: void
-
-  - Response payload
-
-    type: object
-    ```typescript
-    {
-      output: [
-        {
-          name: string;
-          value: any;
-        }
-      ]
-    }
-    ```
-
-  ### Generic events
+### PLUGIN SPECIFIC API
+
+ShellSDK provide a set of features which are specifically designed to allow communications with plugins running inside an application as part of the extention feature.
+
+  - #### VIEW STATE : an all instance synced data object
+
+	You might need to share between you applicatin and plugins a general context to provide coherent UI. ShellSDK let you share any `{ key: value }` object through the ViewState entity. You can define a key from any application or any plugin using the `setViewState` method. ViewState is not persistant and will be deleted when user navigate outside of the application.
+	
+	``` typescript
+	this.sdk.setViewState('TECHNICIAN', id);
+	``` 
+	
+	To listen on modification event, use the listenner `onViewState` with the expected key.
+	
+	```typescript
+	this.sdk.onViewState('TECHNICIAN', id => {
+	    this.selectedId = id;
+	}))
+	```
+	
+	A full instance of the ViewState is stored in the shell and provided on the `REQUEST_CONTEXT` event for initialisation.

```

Comment:
Plugin app may require only slice of the state, why we need to provide everything with context ? My opinion that only way to get viewState should be subscription with `onViewState` method.

Reply:
True, phrasing is pretty bad here. When sending a request_context message, plugin receive an event then receive individual values on corresponding listener.